tests/dragonfly/cluster_test.py

```diff
@@ -396,3 +396,176 @@ async def test_cluster_replica_sets_non_owned_keys(df_local_factory):
+              {{
+                "id": "{replica_ids[1]}",
+                "ip": "localhost",
+                "port": {replicas[1].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 10001,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[2]}",
+            "ip": "localhost",
+            "port": {masters[2].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[2]}",
+                "ip": "localhost",
+                "port": {replicas[2].port}
+              }}
+          ]
+        }}
+      ]
+    """
+    await push_config(config, c_masters_admin + c_replicas_admin)
+
+    client = aioredis.RedisCluster(decode_responses=True, host="localhost", port=masters[0].port)
+
+    for i in range(100):
+        key = 'key' + str(i)
+        assert await client.set(key, 'value') == True
+        assert await client.get(key) == 'value'
+
+    # Make sure that getting a value from a replica works as well.
+    replica_response = await client.execute_command(
+            'get', 'key0', target_nodes=aioredis.RedisCluster.REPLICAS)
+    assert 'value' in replica_response.values()
+
+    # Push new config
+    config = f"""
+      [
+        {{
+          "slot_ranges": [
+            {{
+              "start": 0,
+              "end": 4000
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[0]}",
+            "ip": "localhost",
+            "port": {masters[0].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[0]}",
+                "ip": "localhost",
+                "port": {replicas[0].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 4001,
+              "end": 14000
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[1]}",
+            "ip": "localhost",
+            "port": {masters[1].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[1]}",
+                "ip": "localhost",
+                "port": {replicas[1].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 14001,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[2]}",
+            "ip": "localhost",
+            "port": {masters[2].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[2]}",
+                "ip": "localhost",
+                "port": {replicas[2].port}
+              }}
+          ]
+        }}
+      ]
+    """
+    await push_config(config, c_masters_admin + c_replicas_admin)
+
+    for i in range(100):
+        key = 'key' + str(i)
```

Comment:
I think that if you want to set only 100 keys you should generate a random key so that we test all slots when this test runs few times